js/src/components/contact-information/phone-number-card/phone-number-card.test.js

```diff
@@ -0,0 +1,262 @@
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom';
+import { screen, render, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { parsePhoneNumberFromString as parsePhoneNumber } from 'libphonenumber-js';
+
+/**
+ * Internal dependencies
+ */
+import PhoneNumberCard from './phone-number-card';
+
+jest.mock( '@woocommerce/components', () => ( {
+	...jest.requireActual( '@woocommerce/components' ),
+	Spinner: jest
+		.fn( () => <div role="status" aria-label="spinner" /> )
+		.mockName( 'Spinner' ),
+} ) );
+
+jest.mock( '.~/hooks/useCountryKeyNameMap', () =>
+	jest
+		.fn()
+		.mockReturnValue( { US: 'United States' } )
+		.mockName( 'useCountryKeyNameMap' )
+);
+
+jest.mock( '.~/data', () => ( {
+	...jest.requireActual( '.~/data' ),
+	useAppDispatch() {
+		return {
+			requestPhoneVerificationCode: jest
+				.fn( () => Promise.resolve( { verificationId: 123 } ) )
+				.mockName( 'requestPhoneVerificationCode' ),
+
+			verifyPhoneNumber: jest
+				.fn( () => Promise.resolve( {} ) )
+				.mockName( 'verifyPhoneNumber' ),
+		};
+	},
+} ) );
+
+describe( 'PhoneNumberCard', () => {
+	let phoneData;
+	let phoneNumber;
+
+	function mockPhoneData( fullPhoneNumber ) {
+		const parsed = parsePhoneNumber( fullPhoneNumber );
+
+		phoneData = {
+			...parsed,
+			display: parsed.formatInternational(),
+			isValid: parsed.isValid(),
+		};
+
+		phoneNumber = {
+			...phoneNumber,
+			data: phoneData,
+		};
+	}
+
+	function mockVerified( isVerified ) {
+		phoneNumber = {
+			...phoneNumber,
+			data: {
+				...phoneData,
+				isVerified,
+			},
+		};
+	}
+
+	function mockLoaded( loaded ) {
+		phoneNumber = {
+			...phoneNumber,
+			loaded,
+		};
+	}
+
+	beforeEach( () => {
+		mockPhoneData( '+12133734253' );
+		mockVerified( true );
+		mockLoaded( true );
+	} );
+
+	it( 'When not yet loaded, should render a loading spinner', () => {
+		mockLoaded( false );
+
+		render( <PhoneNumberCard phoneNumber={ phoneNumber } /> );
+
+		const spinner = screen.getByRole( 'status', { name: 'spinner' } );
+		const display = screen.queryByText( phoneData.display );
+		const button = screen.queryByRole( 'button' );
+
+		expect( spinner ).toBeInTheDocument();
+		expect( display ).not.toBeInTheDocument();
+		expect( button ).not.toBeInTheDocument();
+	} );
+
+	it( 'When `initEditing` is not specified, should render in display mode after loading a verified phone number', () => {
```

Comment:
"display mode"  doesn't explain much. I guess it would be more accurate to say 

should render the edit button after loading a verified phone number

Reply:
The mode for entering verification code also has an edit button. Changed to use "non-editing mode" in https://github.com/woocommerce/google-listings-and-ads/pull/2027/commits/ca7ce02bbb5aa9e407dfe26ac47ecff3be8344a9.

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/c9a97fd5-0b03-4f72-8b72-dd596bf620c5)